bliss.js

```diff
@@ -483,57 +507,43 @@ $.setProps = {
 		}
 	},
 
-	once: function(val) {
-		if (arguments.length == 2) {
-			val = {};
-			val[arguments[0]] = arguments[1];
-		}
-
+	once: overload(function(events, callback){
+		events = events.split(/\s+/);
 		var me = this;
+		var once = function() {
+			events.forEach(function(event){
+				me.removeEventListener(event, once);
+			});
 
-		$.each(val, function(events, callback){
-			events = events.split(/\s+/);
-
-			var once = function() {
-				events.forEach(function(event){
-					me.removeEventListener(event, once);
-				});
-
-				return callback.apply(me, arguments);
-			};
+			return callback.apply(me, arguments);
+		};
 
-			events.forEach(function (event) {
-				me.addEventListener(event, once);
-			});
+		events.forEach(function (event) {
+			me.addEventListener(event, once);
 		});
-	},
+	}, 0),
 
 	// Event delegation
-	delegate: function(val) {
-		if (arguments.length === 3) {
-			// Called with ("type", "selector", callback)
-			val = {};
-			val[arguments[0]] = {};
-			val[arguments[0]][arguments[1]] = arguments[2];
-		}
-		else if (arguments.length === 2) {
-			// Called with ("type", selectors & callbacks)
-			val = {};
-			val[arguments[0]] = arguments[1];
-		}
-
-		var element = this;
-
-		$.each(val, function (type, callbacks) {
-			element.addEventListener(type, function(evt) {
-				for (var selector in callbacks) {
-					if (evt.target.closest(selector)) {
-						callbacks[selector].call(this, evt);
-					}
+	delegate: overload(function (type, selector, callback) {
```

Comment:
Hi Lea,

You could also keep the overload() function as it is,  and invoke it twice with different start values, like this : 
```javascript 
delegate: overload( overload(function(type,selector,callback){ ....  },1) ,0);
```


dirk